File: examples/leetcode-out/swift/3/longest-substring-without-repeating-characters.swift
import Foundation

func _indexString(_ s: String, _ i: Int) -> String {
	var idx = i
	let chars = Array(s)
	if idx < 0 { idx += chars.count }
	if idx < 0 || idx >= chars.count { fatalError("index out of range") }
	return String(chars[idx])
}

func lengthOfLongestSubstring(_ s: String) -> Int {
	let s = s
	
	let n = s.count
	var start = 0
	var best = 0
	var i = 0
	while i < n {
		var j = start
		while j < i {
			if _indexString(s, j) == _indexString(s, i) {
				start = j + 1
				break
			}
			j = j + 1
		}
		let length = i - start + 1
		if length > best {
			best = length
		}
		i = i + 1
	}
	return best
}

func main() {
}
main()

File: examples/leetcode-out/cpp/3/longest-substring-without-repeating-characters.cpp
#include <bits/stdc++.h>
using namespace std;

string _indexString(const string& s, int i) {
	int n = s.size();
	if (i < 0) i += n;
	if (i < 0 || i >= n) throw std::out_of_range("index out of range");
	return string(1, s[i]);
}

int lengthOfLongestSubstring(string s){
	auto n = s.size();
	auto start = 0;
	auto best = 0;
	auto i = 0;
	while (i < n) {
		auto j = start;
		while (j < i) {
			if (_indexString(s, j) == _indexString(s, i)) {
				start = j + 1;
				break;
			}
			j = j + 1;
		}
		auto length = i - start + 1;
		if (length > best) {
			best = length;
		}
		i = i + 1;
	}
	return best;
}

int main() {
	return 0;
}
